src/frameworks/next/index.ts

```diff
@@ -210,10 +246,37 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
       }
     }
 
-    const prerenderManifestBuffer = await readFile(
-      join(sourceDir, distDir, "prerender-manifest.json")
-    );
+    const [prerenderManifestBuffer, routesManifestBuffer] = await Promise.all([
+      readFile(
+        join(
+          sourceDir,
+          distDir,
+          "prerender-manifest.json" // TODO: get this from next/constants
+        )
+      ),
+      readFile(
+        join(
+          sourceDir,
+          distDir,
+          "routes-manifest.json" // TODO: get this from next/constants
+        )
+      ),
+    ]);
+
     const prerenderManifest = JSON.parse(prerenderManifestBuffer.toString());
+    const routesManifest = JSON.parse(routesManifestBuffer.toString()) as Manifest;
```

Comment:
Is it possible `routesManifestBuffer` or `prerenderManifestBuffer` is not JSON.parse-able?

Should we try-catch guard this to prevent errors?

Reply:
probably better to refactor to readJSON & throw rather than get too crazy IMO